config/management/operational/src/governance.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_global_constants::LIBRA_ROOT_KEY;
+use libra_management::json_rpc::JsonRpcClientWrapper;
+use libra_management::TransactionContext;
+use libra_management::{
+    constants, error::Error, secure_backend::ValidatorBackend, storage::StorageWrapper,
+};
+use libra_secure_time::{RealTimeService, TimeService};
+use libra_types::account_address::AccountAddress;
+use libra_types::account_config::libra_root_address;
+use libra_types::chain_id::ChainId;
+use libra_types::transaction::RawTransaction;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct AddValidator {
+    #[structopt(long, help = "JSON-RPC Endpoint (e.g. http://localhost:8080")]
+    host: String,
+    #[structopt(long)]
+    chain_id: ChainId,
+    #[structopt(long, help = "The validator address to be added")]
+    address_to_add: AccountAddress,
+    #[structopt(flatten)]
+    libra_root: ValidatorBackend,
+}
+
+impl AddValidator {
+    pub fn execute(self) -> Result<TransactionContext, Error> {
+        let client = JsonRpcClientWrapper::new(self.host);
+        if let Err(e) = client.validator_config(self.address_to_add) {
+            return Err(Error::JsonRpcReadError("validator-config", e.to_string()));
+        }
```

Comment:
I'm pretty sure the error coming out of `validator_config` is the correct error at this point and you should be able to just put a `?` at the end.